### PR TITLE
Add support for scheduled emails

### PR DESCRIPTION
**Pull request by ondrejcernos**

### CODE REVIEW

**Comment by martiner** [comment]
ScheduledMailBuilder/ScheduledMail - what are types of `to` and `bcc`? Email addresses? Please add javadocs and consider to rename it to something more descriptive. Like `toAddress`?

File: src/main/java/com/gooddata/md/ScheduledMailBuilder.java [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+    public ScheduledMailBuilder addAttachment(Attachment att) {
+        if (null == attachments) {
+            attachments = new ArrayList<>();
+        }
+        attachments.add(att);
+        return this;
+    }
+
+    public ScheduledMailBuilder addReportAttachment(String uri, Map<String, String> exportOptions, String... formats) {
```
Comment (martiner) [inline src/main/java/com/gooddata/md/ScheduledMailBuilder.java]: I would use the type `Report` or `ReportDefinition` rather than plain uri.

**Comment by martiner** [comment]
I would use some javadoc for `recurrency`

**Comment by martiner** [comment]
- [ ] add some info into readme

File: src/main/java/com/gooddata/md/ScheduledMail.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    @JsonIgnoreProperties(ignoreUnknown = true)
+    private static class Content {
+
+        @JsonProperty("when")
+        private final ScheduledMailWhen scheduledMailWhen;
+        private final Collection<String> to;
+        private final Collection<String> bcc;
+        private final String subject;
+        private final String body;
+        private final Collection<? extends Attachment> attachments;
```
Comment (martiner) [inline src/main/java/com/gooddata/md/ScheduledMail.java]: why wildcard?

**Comment by martiner** [comment]
- [ ] Please merge ScheduledMailBuilder with ScheduledMail (with retaining handy `add*` methods etc)

File: src/main/java/com/gooddata/md/ScheduledMail.java [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+        super(new Meta(null, null, null, null, summary, title, null, "", null, false, null, false, false));
+        this.content = new Content();
+    }
+
+    /**
+     * Creates full, safe mail schedule object.
+     *
+     * @param title the title of the MD object
+     * @param summary the summary of the MD object
+     * @param recurrency cron-like recurrency of the mail schedule
```
Comment (martiner) [inline src/main/java/com/gooddata/md/ScheduledMail.java]: is it really cron-like? :)

Comment (ondrejcernos) [inline src/main/java/com/gooddata/md/ScheduledMail.java]: it depends on how you define likeness

**Comment by martiner** [comment]
close #158

**Pull request merged**